### PR TITLE
fix: authentication not respecting ignoreUnknownCa and API flag [ROAD-598]

### PR DESCRIPTION
**Pull request by michelkaporin**

### CODE REVIEW

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/CLI/SnykCli.cs [hunk 1]
```diff
@@ -79,46 +96,52 @@ public ISnykOptions Options
...
-            Logger.Information("Adding token");
+            return ConvertRawCliStringToCliResult(consoleResult);
+        }
 
+        public StringDictionary BuildScanEnvironmentVariables()
+        {
+            var environmentVariables = new StringDictionary();
             if (!string.IsNullOrEmpty(this.Options.ApiToken))
             {
+                environmentVariables.Add("SNYK_TOKEN", this.Options.ApiToken);
```
Comment (aldanchenko) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/CLI/SnykCli.cs:119]: Just a suggestion: Maybe use this `ApiEnvironmentVariableName`?

**Review by aldanchenko — APPROVED** [review]
Great work 🚀! it's a pleasure for me to see your С# code 😀

**Comment by aldanchenko** [comment]
@michelkaporin one question that came to mind: maybe we need to fix it main branch? Because this issue related to stable version.

**Comment by michelkaporin** [comment]
@aldanchenko we plan to release this branch in the current sprint. I don't think it's worth investing the time to update the old main branch, since we'll merge `snykcode-dev` in.

**Pull request merged**